opendevin/events/stream.py

```diff
@@ -0,0 +1,31 @@
+import asyncio
+from datetime import datetime
+from typing import Callable, Dict, List
+
+from .event import Event
+
+
+class EventStream:
+    _subscribers: Dict[str, Callable] = {}
+    _events: List[Event] = []
+    _lock = asyncio.Lock()
+
+    def subscribe(self, id: str, subscriber: Callable):
+        if id in self._subscribers:
+            raise ValueError('Subscriber already exists: ' + id)
```

Comment:
This could show an error in the flow, so we should log it to track it down. I don't know if we need to be so harsh to raise though, it's just adding an event listener. In general, it's at least possible, if not likely, to continue normally on something like this

Reply:
Great point!

Reply:
Fixed